src/Executor/ReleaseGroup/ReleaseGroupManifestExecutor.php

```diff
@@ -0,0 +1,160 @@
+     */
+    protected const MASTER_BRANCH_NAME = 'master';
+
+    /**
+     * @var string
+     */
+    protected const INTEGRATOR_RESULT_BRANCH_NAME = 'integrator/release-group-manifest-run';
+
+    /**
+     * @var string
+     */
+    protected const DIFF_TO_DISPLAY_FILE_NAME = 'diff_to_display.diff';
+
+    /**
+     * @var \SprykerSdk\Integrator\Manifest\FileBucketManifestReaderInterface
+     */
+    protected FileBucketManifestReaderInterface $manifestReader;
+
+    /**
+     * @var \SprykerSdk\Integrator\FileStorage\BucketFileStorageInterface
+     */
+    protected BucketFileStorageInterface $bucketFileStorage;
+
+    /**
+     * @var \SprykerSdk\Integrator\Executor\ManifestExecutorInterface
+     */
+    protected ManifestExecutorInterface $manifestExecutor;
+
+    /**
+     * @var \SprykerSdk\Integrator\VersionControlSystem\GitRepository
+     */
+    protected GitRepository $gitRepository;
+
+    /**
+     * @param \SprykerSdk\Integrator\Manifest\FileBucketManifestReaderInterface $manifestReader
+     * @param \SprykerSdk\Integrator\FileStorage\BucketFileStorageInterface $bucketFileStorage
+     * @param \SprykerSdk\Integrator\Executor\ManifestExecutorInterface $manifestExecutor
+     * @param \SprykerSdk\Integrator\VersionControlSystem\GitRepository $gitRepository
+     */
+    public function __construct(
+        FileBucketManifestReaderInterface $manifestReader,
+        BucketFileStorageInterface $bucketFileStorage,
+        ManifestExecutorInterface $manifestExecutor,
+        GitRepository $gitRepository
+    ) {
+        $this->manifestExecutor = $manifestExecutor;
+        $this->bucketFileStorage = $bucketFileStorage;
+        $this->manifestReader = $manifestReader;
+        $this->gitRepository = $gitRepository;
+    }
+
+    /**
+     * @param int $releaseGroupId
+     * @param \SprykerSdk\Integrator\Dependency\Console\InputOutputInterface $inputOutput
+     * @param \SprykerSdk\Integrator\Transfer\IntegratorCommandArgumentsTransfer $commandArgumentsTransfer
+     *
+     * @throws \RuntimeException
+     *
+     * @return int
+     */
+    public function runReleaseGroupManifestExecution(
+        int $releaseGroupId,
+        InputOutputInterface $inputOutput,
+        IntegratorCommandArgumentsTransfer $commandArgumentsTransfer
+    ): int {
+        $manifests = $this->manifestReader->readManifests($releaseGroupId);
+        $unappliedManifests = $this->manifestExecutor->findUnappliedManifests($manifests, []);
+        if (!count($unappliedManifests)) {
+            throw new RuntimeException(
+                sprintf('No unapplied manifests found for release group id: %s', $releaseGroupId),
+            );
+        }
+
+        $dry = $commandArgumentsTransfer->getIsDryOrFail();
+        if (!$dry) {
+            $this->prepareBranch();
+        }
+
+        $this->manifestExecutor->applyManifestList($unappliedManifests, $inputOutput, $commandArgumentsTransfer);
+
+        if ($dry) {
+            return 0;
```

Comment:
what does it mean 0? if all cases 0 it means void

Reply:
We should also not leak CLI topic (exit codes) with internal code (bool or otherwise)

Reply:
thanks ++

Reply:
Thanks, changed to void